api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1381,25 +1382,106 @@ public void getObject(String bucketName, String objectName, String fileName)
     }
   }
 
+  /**
+   * Copy a source object into a new object with the provided name in the provided bucket.
+   *
+   * </p>
+   * <b>Example:</b><br>
+   * 
+   * <pre>
+   * {@code minioClient.copyObject("my-bucketname", "my-objectname", "/my-sourcebucketname/my-objectsourcename", 
+   * copyConditions); }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name to be created after copy.
+   * @param objectSource
+   *          Object name of the source.
+   * @param copyConditions
+   *          CopyConditions object with.
+   *
+   * @throws InvalidBucketNameException
+   *           upon an invalid bucket name
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   */
+  public void copyObject(String bucketName, String objectName, String objectSource, CopyConditions copyConditions)
+      throws InvalidKeyException, InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException,
+      NoResponseException, ErrorResponseException, InternalException, IOException, XmlPullParserException,
+      InvalidArgumentException {
+    if (copyConditions == null) {
+      throw new InvalidArgumentException("Copy conditions should be a non null value");
+    }
+    if (objectSource == null) {
+      throw new InvalidArgumentException("Invalid object source");
+    }
+    String sourceBucketName = getBucketNameFromPath(objectSource);
+    String sourceObjectName = getObjectNameFromPath(objectSource);
+    ObjectStat objectStat = statObject(sourceBucketName, sourceObjectName);
+    String etag = objectStat.etag();
+    Date lastModified = objectStat.createdTime();
+    long length = objectStat.length();
+
+    // If the conditions specified are satisfied, get and put the object in new location
+    if (copyConditions.isUnmodifiedSince(lastModified) && copyConditions.isModifiedAfter(lastModified)
+        && copyConditions.etagMatches(etag) && copyConditions.etagDoesNotMatch(etag)) {
+      InputStream objectByteStream = getObject(sourceBucketName, sourceObjectName);
+      putObject(bucketName, objectName, objectByteStream, length, "application/octet-stream");
```

Comment:
https://docs.minio.io/docs/python-client-api-reference#copy_object - here you go. 

https://github.com/minio/minio-py/blob/master/minio/api.py#L769 - For the reply you need to parse CopyObjectResult XML.